server/dive_server/utils.py

```diff
@@ -101,3 +114,17 @@ def saveTracks(folder, tracks, user):
         user=user,
         mimeType="application/json",
     )
+
+
+def saveAttributes(folder, attributes, user):
+    attributes_dict = {}
+    if 'attributes' in folder['meta']:
+        attributes_dict = folder['meta']['attributes']
+    # we dont overwrite any existing meta attributes
+    for attribute in attributes.values():
+        validated: models.Attribute = models.Attribute(**attribute)
+        if attribute['key'] not in attributes_dict:
+            attributes_dict[str(validated.key)] = validated.dict(exclude_none=True)
+
+    folder['meta']['attributes'] = attributes_dict
+    Folder().save(folder)
```

Comment:
can this be refactored so that both `viame.postprocess` and `viame.save_attributes` can call the same save utility function?

Reply:
They can but there isn't really much overlap besides the two lines for the setting of the `folder['meta']['attributes']` and the saving of the folder.
`viame.save_attributes` - utilizes the upsert/delete lists for input and does overwriting of the current attribute definitions
`utils.saveAttributes` takes an attributes dict directly and won't overwrite any existing attribute definitions.

It may be better for me to clarify this function as something more like `saveCSVImportAttributes`?

Reply:
You have a comment here that says "We don't overwrite existing" but since this is only called in the `postProcess` function, I can't come up with a case where there were any attributes on the folder to begin with.  How might there be existing attributes?

Reply:
Post process is called on any pipeline runs as well.  We don't have pipelines that create any attributes but I don't know if that will always be true.  We can confirm with Matt but I'm guessing at some point the that people won't want user input pipelines like box->polys to wipe out their set attributes when run like they do now.

Reply:
Ahh, that's what I was forgetting.  Thanks.   Function rename is fine with me.